### PR TITLE
Improvements to the README

### PR DESCRIPTION
**Pull request by pwaller**
This fleshes out the README considerably. Comments welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,24 +16,142 @@ such as when a branch is pushed to.
 This is useful for creating a continuous deployment infrastructure, which runs tests
 or deploys software when it is updated.
 
+## The problem with web hooks
+
 The problem with webhooks is that it requires that the receiver of the hook listens on
```
Comment (StevenMaude) [inline README.md]: This needs plural agreement: "The problem with webhooks are that they require that..."

Optionally, you could simplify it by "Webhooks require that ...": you've already said in the header that they are a problem.

File: README.md [hunk 2]
```diff
@@ -16,24 +16,142 @@ such as when a branch is pushed to.
...
+Hookbot has two types of URL, `https://uri-specific-token@host/pub/<channel>` and `wss://uri-specific-token@host/sub/<channel>`.
+
+Webhooks are broadcast from a HTTP POST to `/pub/` and received by making a websocket
+connection to a `/sub/` URI and reading messages from it.
+
+The authentication is a hmac of a single secret with the URI, so a token used for
+publishing is different from the one for listening; tokens are also different
+between separate channels. The single secret is only known by the server.
+
+Hookbot is licensed under a BSD like license.
```
Comment (StevenMaude) [inline README.md]: Maybe the licensing should go at the start or end?

File: README.md [hunk 3]
```diff
@@ -16,24 +16,142 @@ such as when a branch is pushed to.
...
+Instead, we can configure
+[GitHub's organization level webhook](https://developer.github.com/v3/orgs/hooks/) 
+once. The problem then is that every project gets details of every other project's
+hooks, which would not be ideal. To avoid this, hookbot can rebroadcast messages
+from one channel onto other channels.
+
+For example, let's say `scraperwiki/hookbot`'s `master` branch is updated.
 
-Hookbot
+The `/unsafe/pub/github.com/org/scraperwiki?extra-metadata=github` channel recieves
```
Comment (StevenMaude) [inline README.md]: Small typo: recieves.

**Comment by StevenMaude** [comment]
Apart from the small changes, it's clear what hookbot does and how to use it. Depending on your preference, you might also want to add the Travis badge while you're altering the README.

Other than that, looks good to merge.

**Comment by pwaller** [comment]
I've addressed Steve's comment, someone else can have a skim.

**Pull request merged**